src/components/Pagination.tsx

```diff
@@ -8,24 +8,27 @@ import {
 import { addOpacityToHsla } from "@/helpers";
 import PreviousPage from "public/assets/icons/left-chevron.svg";
 import NextPage from "public/assets/icons/right-chevron.svg";
-import { useEffect, useState } from "react";
+import { useCallback, useEffect, useState } from "react";
 import styled from "styled-components";
 
-interface Props<Entry> {
-  entries: Entry[];
-  setEntriesToShow: (entries: Entry[]) => void;
-  findIndex?: number;
-}
 /**
- * Handles pagination for a list of entries
- * @param entries - the entries to paginate (not the entries to show)
- * @param setEntriesToShow - the function to call when the entries to show change
+ * @description
+ * This component is used to navigate between pages of a list of entries.
+ * @param entries - The list of entries to paginate.
+ * @param findIndex - The index of the entry to find, if any.
+ * @returns
+ * An array of entries to show for the current page.
+ * All of the props needed to render the pagination component.
  */
-export function Pagination<Entry>({
-  entries,
-  setEntriesToShow,
-  findIndex,
-}: Props<Entry>) {
+export function usePagination<Entry>(entries: Entry[], findIndex?: number) {
+  const [entriesToShow, setEntriesToShow] = useState<typeof entries>([]);
```

Comment:
This hook now holds all the pagination state, including the `entriesToShow` state that was previously being re-implemented in components.

Reply:
This logic was the cause of the bug — previously the `useState` used the whole list of entries as its initial value, which caused the component to attempt to render all thousands of them before the effect hook in the pagination could run. Starting off with an empty array fixes this.